### PR TITLE
#161476607 Post a new product 

### PR DESCRIPTION
**Pull request by Georgeygigz**
**What does this PR do?**
Creates an endpoint that allows users to post a new  products

**Description of the completed  task**
 1. Create an Endpoint where a user can post a new products
 2. Refactor the code if test doesnt pass

## END POINTS
| EndPoints       | Functionality  | HTTP Method  |
| ------------- |:-------------:| -----:|
| api/v2/products | Get all the products| GET |
| api/v2/products/<int:product_id> | Fetch single product |GET|
| api/v2/products |Add a new product |POST|

**How to manually test this**
 1. Make a new directory on your computer
 2. Open the terminal and navigate to the folder
 3. `git clone` this  <code>[repo](https://github.com/Georgeygigz/store-manager-api/)</code>
 4.  run `pip install -r requirements.txt` to install the dependencies
 5.  Then on your terminal and run the application
 6. Then on [postman](https://www.getpostman.com/), navigate to this url  to post a new  product   `api/v2/products`

**PT stories**
#161476607 

image

### CODE REVIEW

File: app/api/v2/models/store_model.py [hunk 1]
```diff
@@ -24,3 +24,12 @@ def get_all_products(self):
...
         return all_products
+
+    def insert_new_product(self,  product_id, product_name, category_id, stock_amount, price, low_inventory_stock):
+        database = self.db
+        curr = database.cursor()
+        query = "INSERT INTO products (product_id,product_name,category_id,stock_amount,price,low_inventory_stock) VALUES (%s,%s,%s,%s,%s,%s);"
+        curr.execute(query, (product_id, product_name, category_id,
+                             stock_amount, price, low_inventory_stock))
+        database.commit()
+        return {"Message": "Product Save succefully"}, 201
```
Comment (dotNesh) [inline app/api/v2/models/store_model.py]: Response message should be something like "Product added successfully"

File: app/api/v2/utils/utils.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    def validate_empty_product_inputs(self,data):
+        if (data['product_name'] == "") :
+            abort(400, description="Product name required")
+        if data['stock_amount'] == "":
+            abort(400, description="Stock amount should not be empty")
+        if data['low_inventory_stock'] == "":
+            abort(400, description="low inventory stock field should not be empty")
+        if data['price'] == "":
+            abort(400, description="Price should not be empty")
+
```
Comment (dotNesh) [inline app/api/v2/utils/utils.py]: Leave a line at the end of the file.

File: app/api/v2/models/store_model.py (hunk 1 above)
Comment (Georgeygigz) [reply app/api/v2/models/store_model.py]: Okay, Will check on that

File: app/api/v2/utils/utils.py (hunk 2 above)
Comment (Georgeygigz) [reply app/api/v2/utils/utils.py]: Okay, I will check on that too

**Pull request merged**